.github/workflows/release.yaml

```diff
@@ -20,21 +23,124 @@ jobs:
+      - name: Build and Package SAM Application and Upload to S3
+        env:
+          bucket_name: unified-lambda-serverless
+          region: us-east-2
         run: |
-          pip install aws-sam-cli
+          BUILD_DIR_BASE=".aws-sam/build"
+          TEMPLATES=(
+            "logging-lambda-metric-polling.yaml"
+            "logging-lambda-metric-stream.yaml"
+            "logging-firehose-metric-polling.yaml"
+            "logging-firehose-metric-stream.yaml"
+            "logging-lambda-firehose-metric-polling.yaml"
+            "logging-lambda-firehose-metric-stream.yaml"
+            "lambda-template.yaml"
+            "logging-lambda-firehose-template.yaml"
+          )
+          
+          for TEMPLATE_FILE in "${TEMPLATES[@]}"; do
+            BASE_NAME=$(basename "$TEMPLATE_FILE" .yaml)
+            BUILD_DIR=".aws-sam/build/$BASE_NAME"
+          
+            sam build --template-file "$TEMPLATE_FILE" --build-dir "$BUILD_DIR"
+            sam package --s3-bucket "$bucket_name" --template-file "$BUILD_DIR/template.yaml" --output-template-file "$BUILD_DIR/$TEMPLATE_FILE"  --region $region
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to package template $TEMPLATE_FILE"
+            fi  
+          
+            aws s3 cp "$BUILD_DIR/$TEMPLATE_FILE" "s3://$bucket_name/$TEMPLATE_FILE"
+            if [[ $? -ne 0 ]]; then
+              echo "Error: Failed to copy $TEMPLATE_FILE to S3 bucket $bucket_name"
+            fi
+          
+          done
+
+  deploy-lambda-code-parallelly-to-all-region:
+    needs: build-and-deploy
+    permissions:
+      id-token: write
+      contents: write
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        region:
+          - us-east-1
+          - us-east-2
+          - eu-west-1
+          - eu-west-2
+          - us-west-1
+          - us-west-2
+          - af-south-1
+          - ap-east-1
+          - ap-south-2
+          - ap-southeast-3
+          - ap-southeast-5
+          - ap-southeast-4
+          - ap-south-1
+          - ap-northeast-3
+          - ap-northeast-2
+          - ap-southeast-1
+          - ap-southeast-2
+          - ap-northeast-1
+          - ca-central-1
+          - ca-west-1
+          - eu-central-1
+          - eu-south-1
+          - eu-west-3
+          - eu-south-2
+          - eu-north-1
+          - eu-central-2
+          - me-south-1
+          - me-central-1
+          - il-central-1
+          - sa-east-1
+
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: log-forwarder-zip
+          path: ./build-artifacts
 
-      - name: Build SAM Application
-        run: sam build --template-file template.yaml --region us-east-2
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: ${{ secrets.AWS_S3_PUBLISH_ROLE }}
+          aws-region: us-east-2
```

Comment:
Do we need region here?

Reply:
Assuming role requires region to be provided for temporary credentials creation